core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -123,18 +116,9 @@ static struct unwind_idx *find_index(uint32_t addr, uaddr_t exidx,
 	int32_t prel31_addr;
 	vaddr_t func_addr;
 
-	if (exidx && exidx_sz) {
-		/* Unwinding a user TA */
-		start = (struct unwind_idx *)exidx;
-		idx_start = (vaddr_t)exidx;
-		idx_end = (vaddr_t)(exidx + exidx_sz);
-
-	} else {
-		/* Unwinding the kernel stack: use kernel tables */
-		start = &__exidx_start;
-		idx_start = (vaddr_t)&__exidx_start;
-		idx_end = (vaddr_t)&__exidx_end;
-	}
+	start = (struct unwind_idx *)exidx;
+	idx_start = (vaddr_t)exidx;
```

Comment:
Well, we don't have a clear policy on `uaddr_t` (user space address) versus `vaddr_t` (normal virtual address). When we know that an address can be used directly by OP-TEE OS then it could be represented by a `vaddr_t` since all prerequisites are in place, until those checks hasn't been done a `uaddr_t` should stay `uaddr_t`.

`find_index()` works with both pure kernel-mode pointers but also with user-mode pointers that has prerequisites in place (the correct vaspace is mapped) so this function should work with `vaddr_t` only.